volume/volume_windows.go

```diff
@@ -0,0 +1,163 @@
+package volume
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	derr "github.com/docker/docker/errors"
+)
+
+// read-write modes
+var rwModes = map[string]bool{
+	"rw": true,
+}
+
+// read-only modes
+var roModes = map[string]bool{
+	"ro": true,
+}
+
+const (
+	// Spec should be in the format [source:]destination[:mode]
+	//
+	// Examples: c:\foo bar:d:rw
+	//           c:\foo:d:\bar
+	//           myname:d:
+	//           d:\
+	//
+	// Explanation of this regex! Thanks @thaJeztah on IRC and gist for help. See
+	// https://gist.github.com/thaJeztah/6185659e4978789fb2b2. A good place to
+	// test is https://regex-golang.appspot.com/assets/html/index.html
+	//
+	// Useful link for referencing named capturing groups:
+	// http://stackoverflow.com/questions/20750843/using-named-matches-from-go-regex
+	//
+	// There are three match groups: source, destination and mode.
+	//
+
+	// RXHostDir is the first option of a source
+	RXHostDir = `[a-zA-Z]:\\(?:[^\\/:*?"<>|\r\n]+\\?)*`
+	// RXName is the second option of a source
+	RXName = `[a-zA-Z0-9][a-zA-Z0-9_.-]+`
```

Comment:
Gosh, do you have children? ;) Sure, I can't account for user stupidity on doing something like that should they want, but there's no reason to give them a loaded gun, at least on Windows. There is no way I'm going to allow the floppy-disk icon like name, or volume names such as in https://github.com/docker/docker/pull/16140#issuecomment-141254127. There is absolutely no need to be that open. Remember also that the name equates to a path created under c:\ProgramData\Docker\Volumes so must be valid. So it can't include other characters which are invalid including < > : / \ | ? * https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx. 

I really don't understand your hesitancy here for being restrictive until there is direct end-user feedback that this is not workable - have you any data to say that this is the case? Note also that directory names are case insensitive on NTFS, so it's always lower-cased on Windows anyway - the volume name "John" will equate to "john" during processing - there is no way we can avoid that on Windows.